atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -755,10 +755,14 @@ private void validatePreCommitRequirementsOnReadIfNecessary(TableReference table
     }
 
     private boolean isValidationNecessaryOnReads(TableReference tableRef) {
-        return validateLocksOnReads && isValidationNecessary(tableRef);
+        return validateLocksOnReads && isThoroughSwept(tableRef);
```

Comment:
Not sure if we've set a convention on this, but `isThoroughlySwept` reads better